lib/agent.js

```diff
@@ -424,4 +430,16 @@ function urlPropertyToPacketOption(url, req, property, option, separator) {
       }))
 }
 
+Agent.prototype._convertMulticastToUnicastRequest = function (req, rsinfo) {
+  var reqOpts = {"host": rsinfo.address.split('%')[0], "port": rsinfo.port}
+  var unicast_req = this.request(reqOpts)
+  unicast_req.url = req.url
+  unicast_req.url.host = unicast_req.sender._host
+  unicast_req.url.multicast = false;
+  unicast_req._events = req._events
```

Comment:
Is this needed? I think this might create significant problems as this is internal state of EventEmitter.

Reply:
You are right, this is rather suboptimal. The first three lines weren't actually needed and I removed them in the fixup commits I just added. I also changed the fourth line now to 

```js
unicast_req._events.response = req._events.response
```

so only the function references are copied. Maybe this will work better?

Reply:
(I took the liberty of already squashing the commits.)